build/v1/types.go

```diff
@@ -462,15 +466,32 @@ type ImageSourcePath struct {
 	DestinationDir string `json:"destinationDir" protobuf:"bytes,2,opt,name=destinationDir"`
 }
 
+// LocalObjectBuildSource is a build source that is copied into a build from a Kubernetes
+// key-value store, such as a `Secret` or `ConfigMap`.
+type LocalObjectBuildSource interface {
```

Comment:
we typically don't declare interfaces or methods on API types... if you want to adapt them like this, do so in application code, not in the API structs themselves

Reply:
I wanted to keep this interface within Origin proper, but most of the builder code acts upon API structs and not converted internal representations - see https://github.com/openshift/origin/blob/master/pkg/build/builder/docker.go as an example. Exposing this interface at the API level allows most the code for mounting `Secrets` and `ConfigMaps` to be unified, but may not be the best long-term path.